src/hooks/standardLibs.ts

```diff
@@ -0,0 +1,152 @@
+    assert(id);
+    if (getModuleConfig(id)?.exclude?.includes(key)) return;
+
+    const f = owner[key] as F;
+    if (!proxiedFunctions.has(f)) {
+      const proxy = createProxy(f, id);
+      owner[key] = proxy;
+      proxiedFunctions.add(f);
+    }
+  }
+
+  if (typeof mod === "object" && mod !== null) {
+    setCustomInspect(mod, () => "[" + id + "]");
+    for (const key in mod) {
+      const member = (mod as FunctionOwner)[key];
+      if (typeof member === "function") {
+        wrapFunction(mod as FunctionOwner, key);
+        if (member.prototype && typeof member.prototype == "object") {
+          // If it's a class declaration we should instrument it's methods.
+          // TODO: Clarify if we have to consider inherited methods too?
+          //       Up until Object?
+          const methodNames = getMethodNames(member.prototype as FunctionOwner);
+          for (const methodName of methodNames)
+            wrapFunction(member.prototype as FunctionOwner, methodName);
+        }
+      }
+    }
+    return mod;
+  }
+}
+
+function recordingIsShallowFor(moduleId: string) {
+  return getModuleConfig(moduleId)?.shallow;
+}
+
+function callerIsUserCode() {
+  const fileName = getCallerFile();
+  return !(fileName?.startsWith("node:") ?? fileName?.includes("node_modules"));
+}
+
+function createProxy(f: F, moduleId: string) {
+  const proxy = new Proxy(f, {
+    apply(target, thisArg, argArray: Parameters<F>) {
+      if (recordingIsShallowFor(moduleId) && !callerIsUserCode())
+        return Reflect.apply(target, thisArg, argArray);
+
+      return record.call(thisArg, target, argArray, getFunctionInfo(moduleId, f));
+    },
+  });
+  return proxy;
+}
+
+standardLibsHook.applicable = function (id: string) {
+  return getModuleConfig(id) != undefined;
+};
+
+const functionInfos = new Map<string, FunctionInfo>();
+
+let functionCount = 0;
+function getFunctionInfo(moduleId: string, f: F) {
+  const key = moduleId + "." + f.name;
+
+  if (!functionInfos.has(key)) {
+    const params = Array(f.length).map((i) => {
+      return { type: "Identifier", name: `arg${i}` } as ESTree.Identifier;
+    });
+    const info: FunctionInfo = {
+      async: true,
+      generator: false,
+      id: f.name,
+      params: params,
+      location: { path: `${moduleId}`, lineno: ++functionCount },
+      klassOrFile: moduleId,
+      static: true,
+    };
+    functionInfos.set(key, info);
+  }
+
+  return functionInfos.get(key)!;
+}
+
+function getCallerFile() {
+  const originalFunc = Error.prepareStackTrace;
+
+  let callerFile;
+  try {
+    const err = new Error();
+    Error.prepareStackTrace = (err, stack) => stack;
+
+    const stack = err.stack as unknown as NodeJS.CallSite[];
+    const currentfile = stack.shift()?.getFileName();
+
+    while (stack.length) {
+      callerFile = stack.shift()?.getFileName();
+      if (currentfile !== callerFile) break;
+    }
+  } catch (e) {
+    //
+  }
+
+  Error.prepareStackTrace = originalFunc;
```

Comment:
This is VERY expensive. Remember the library functions will often be in hot loops; this will lead to a massive performance issue.